pilopt/src/lib.rs

```diff
@@ -141,6 +141,26 @@ fn build_poly_id_to_definition_name_lookup(
     poly_id_to_definition_name
 }
 
+/// Builds a lookup-table that can be used to turn array elements
+/// (in form of their poly ids) into the names of the arrays.
+fn build_poly_id_to_array_elem_lookup(
```

Comment:
This is very similar to the function above. Could you please find a way to reduce code duplication? We could have a single lookup tables that also contains a boolean flag, for example.